Feature continuous integration

IntegralProgrammer opened the pull request:
This PR gives the ability to perform automated testing of NCR using my pre-trained models available at https://github.com/IntegralProgrammer/NeuralCR/releases/download/v0.1.1/ncr_model_params.tar.gz

IntegralProgrammer committed:
Data-driven test template prints errors instead of exiting with a non-zero exit code

IntegralProgrammer committed:
Add scripts for data-driven testing

IntegralProgrammer committed:
Add references to the publicly available trained models (HPO, MONDO) in app.py

IntegralProgrammer committed:
Update README.md with info on Continuous Integration

IntegralProgrammer committed:
If AUTOTEST env var is non-null, continuous integration tests will be automatically executed

IntegralProgrammer committed:
If TEST_IGNORE_SCORE env var is non-null, when continuous integration tests are executed, `score` parameter will be ignored when checking for equality of actual and expected result

IntegralProgrammer committed:
Add data for data driven test cases

sdumitriu commented on README.md:
````diff
@@ -176,6 +176,64 @@ curl http://127.0.0.1:5000/annotate/?text=The+paitient+was+diagnosed+with+both+c
+
+```bash
+cd ~
+wget https://github.com/IntegralProgrammer/NeuralCR/releases/download/v0.1.1/ncr_model_params.tar.gz
````
Should point to the official repo.

IntegralProgrammer replied:
Fixed by commit `f28895a` :heavy_check_mark:

sdumitriu commented on download_trained_models.sh:
```diff
@@ -0,0 +1,11 @@
+cd /root/opt/ncr/
+mkdir model_params
+cd model_params
+wget https://github.com/IntegralProgrammer/NeuralCR/releases/download/v0.1.1/ncr_model_params.tar.gz
```
Official repo.

IntegralProgrammer replied:
Fixed by commit `f28895a` :heavy_check_mark:

sdumitriu commented on tests/ci_tests/data_driven/2/url:
```diff
@@ -0,0 +1 @@
+http://127.0.0.1:5000/annotate/?text=The+paitient+was+diagnosed+with+both+cardiac+disease+and+renal+cancer.&model=MONDO
```
Is the `pa_i_tient` typo on purpose, or can it be fixed? Also in several other places.

IntegralProgrammer replied:
Fixed by commit `ba756b0`. Data-driven test results are also updated as `start` and `end` values have changed. :heavy_check_mark: The other spelling mistakes will be fixed in other (non-CI) places in a different PR.

sdumitriu commented on docker_run_webapp.sh:
```diff
@@ -4,4 +4,6 @@ echo "Starting Docker..."
 docker run --rm \
 	-v $(realpath ~/ncr_model_params):/root/opt/ncr/model_params:ro \
```
The next file isn't needed if you do this instead:

```suggestion
	${LOCALMOUNT:+ -v $(realpath ~/ncr_model_params):/root/opt/ncr/model_params:ro} \
```

The above works by only mounting if the `LOCALMOUNT` env is set.

IntegralProgrammer replied:
Done - this change was implemented and the README.md file was updated accordingly :heavy_check_mark:

IntegralProgrammer committed:
Getting rid of need for docker_run_webapp_nomount.sh file.

Instead, docker_run_webapp.sh is used along with the LOCALMOUNT environment variable

IntegralProgrammer committed:
Fix incorrect spelling of 'paitient' to correct spelling - 'patient' in CI tests.

Add revised test results

IntegralProgrammer committed:
Change URL for downloading pre-trained NCR models
